src/screens/SendTokens.js

```diff
@@ -492,8 +497,14 @@ function SendTokens() {
     return <ul>{rows}</ul>
   }
 
-  const renderAlertBody = () => {
-    if (ledgerStep === 0) {
+  /**
+   * Renders the body for the ledger alert modal, with instructions on how to approve the tx.
+   * When called with the `sendTransaction` parameter, it renders a transaction success monitoring
+   * component instead, using this parameter as a reference for which transaction to listen to.
+   * @param {SendTransaction} [sendTransactionObj] Reference to tx already signed by Ledger
+   */
+  const renderAlertBody = (sendTransactionObj) => {
+    if (!sendTransactionObj) {
```

Comment:
suggestion: Use a number/enum/boolean argument instead of a `SendTransaction` instance.

I understand why using `sendTransactionObj` is nice, since the `else` block requires it, but the object is created before we send the data to the Ledger device it is just not passed to the method.
We could have both as arguments `(ledgerStep, sendTransactionObj)`, this way the caller would be in control of the rendered content.

Reply:
Done on https://github.com/HathorNetwork/hathor-wallet/pull/546/commits/5fa10de002266ca6a137076c4f49c7df69b0fc81